src/components/date-range/index.js

```diff
@@ -0,0 +1,56 @@
+import React, { forwardRef } from 'react'
+import PropTypes from 'prop-types'
+
+import './date-range.css'
+
+
+const DateRange = forwardRef(({ classes, defaultValue, setFromValue, setToValue, showLabel }, ref) => {
+  const dateRangeClasses = Object.freeze({
+    form: `${classes.form ? classes.form : 'container w-96 flex'}`,
+    field: `${classes.form ? classes.field : 'w-full mx-2'}`,
+    label: `${classes.form ? classes.label : 'text-xs text-gray-500'}`,
+    input: `${classes.form ? classes.input : 'w-full pb-2 text-sm border-b border-gray-500 focus:outline-none'}`,
+  })
+
+  return (
+    <form ref={ref} className={`${dateRangeClasses.form}`} noValidate>
+      <div className={`fieldContainer ${dateRangeClasses.field}`}>
+        {showLabel && <label className={`${dateRangeClasses.label}`} htmlFor="from">From</label>}
+        <input className={`date-input ${dateRangeClasses.input}`} type="date" id="from" name="from"
+          value={defaultValue[0] || ''} onChange={({ target: { value } }) => setFromValue(value)}/>
+      </div>
+      <div className={`fieldContainer ${dateRangeClasses.field}`}>
+        {showLabel && <label className={`${dateRangeClasses.label}`} htmlFor="to">To</label>}
+        <input className={`date-input ${dateRangeClasses.input}`} type="date" id="to" name="to"
+          value={defaultValue[1] || ''} onChange={({ target: { value } }) => setToValue(value)}/>
```

Comment:
it may be worth documenting the `value` here in the form of `yyyy-mm-dd`, which may be different from what the user sees in the display which goes by the browser locale setting

see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#handling_browser_support for more context and perhaps confine it using the `pattern` attribute